From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Outfit7TSB <sunnybunnies09@naver.com>
Date: Fri, 24 Jan 2025 21:42:33 +0900
Subject: [PATCH] changes


diff --git a/build.gradle.kts b/build.gradle.kts
index 07be173eb4ce57b74090729bb56320f87da2a71f..c0d5c1ed35049a1034a465498a9a8598a18ebb93 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -26,7 +26,6 @@ abstract class MockitoAgentProvider : CommandLineArgumentProvider {
 
 dependencies {
     implementation(project(":NogyangSpigot-API")) // 버킷-스피곳-페이퍼-레인보우 통합 API
-	implementation(project(":Sponge-API")) // 스펀지 API
     implementation("ca.spottedleaf:concurrentutil:0.0.3") // Paper - Add ConcurrentUtil dependency
     // Paper start
     implementation("org.jline:jline-terminal-ffm:3.27.1") // use ffm on java 22+
@@ -43,7 +42,7 @@ dependencies {
     log4jPlugins.annotationProcessorConfigurationName("org.apache.logging.log4j:log4j-core:2.19.0") // Paper - Needed to generate meta for our Log4j plugins
     runtimeOnly(log4jPlugins.output)
     alsoShade(log4jPlugins.output)
-    implementation("io.netty:netty-codec-haproxy:4.1.97.Final") // Paper - Add support for proxy protocol
+    implementation("io.netty:netty-codec-haproxy:4.1.115.Final") // Paper - Add support for proxy protocol
     // Paper end
     implementation("org.apache.logging.log4j:log4j-iostreams:2.24.1") // Paper - remove exclusion
     implementation("org.ow2.asm:asm-commons:9.7.1")
@@ -106,7 +105,7 @@ tasks.jar {
             "Implementation-Vendor" to date, // Paper
             "Specification-Title" to "NogyangSpigotCraftBukkit",
             "Specification-Version" to project.version,
-            "Specification-Vendor" to "Paper Team",
+            "Specification-Vendor" to "Approximaster Studios Two Thousand FOUR",
             "Brand-Id" to "approximasterstudiosserverteam:nogyangspigotcraftbukkit",
             "Brand-Name" to "NogyangSpigotCraftBukkit",
             "Build-Number" to (build ?: ""),
@@ -215,13 +214,13 @@ val runtimeClasspathWithoutVanillaServer = configurations.runtimeClasspath.flatM
         runtime.filterNot { it.asFile.absolutePath == vanilla }
     }
 
-tasks.registerRunTask("runServer") {
+tasks.registerRunTask("runSubJar") {
     description = "Spin up a test server from the Mojang mapped server jar"
     classpath(tasks.includeMappings.flatMap { it.outputJar })
     classpath(runtimeClasspathWithoutVanillaServer)
 }
 
-tasks.registerRunTask("runReobfServer") {
+tasks.registerRunTask("runMainJar") {
     description = "Spin up a test server from the reobfJar output jar"
     classpath(tasks.reobfJar.flatMap { it.outputJar })
     classpath(runtimeClasspathWithoutVanillaServer)
@@ -233,22 +232,22 @@ tasks.registerRunTask("runDevServer") {
     jvmArgs("-DPaper.pushPaperAssetsRoot=true")
 }
 
-tasks.registerRunTask("runBundler") {
+tasks.registerRunTask("runSubBundler") {
     description = "Spin up a test server from the Mojang mapped bundler jar"
     classpath(rootProject.tasks.named<io.papermc.paperweight.tasks.CreateBundlerJar>("createMojmapBundlerJar").flatMap { it.outputZip })
     mainClass.set(null as String?)
 }
-tasks.registerRunTask("runReobfBundler") {
+tasks.registerRunTask("runMainBundler") {
     description = "Spin up a test server from the reobf bundler jar"
     classpath(rootProject.tasks.named<io.papermc.paperweight.tasks.CreateBundlerJar>("createReobfBundlerJar").flatMap { it.outputZip })
     mainClass.set(null as String?)
 }
-tasks.registerRunTask("runPaperclip") {
+tasks.registerRunTask("runSubPaperclip") {
     description = "Spin up a test server from the Mojang mapped Paperclip jar"
     classpath(rootProject.tasks.named<io.papermc.paperweight.tasks.CreatePaperclipJar>("createMojmapPaperclipJar").flatMap { it.outputZip })
     mainClass.set(null as String?)
 }
-tasks.registerRunTask("runReobfPaperclip") {
+tasks.registerRunTask("runMainPaperclip") {
     description = "Spin up a test server from the reobf Paperclip jar"
     classpath(rootProject.tasks.named<io.papermc.paperweight.tasks.CreatePaperclipJar>("createReobfPaperclipJar").flatMap { it.outputZip })
     mainClass.set(null as String?)
