From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Outfit7TSB <sunnybunnies09@naver.com>
Date: Tue, 18 Feb 2025 02:25:19 +0900
Subject: [PATCH] Revolutionize More! Revolution For SpigotMC! Down for @md-5!


diff --git a/build.gradle.kts b/build.gradle.kts
index 7eeac404489e1a813cd461b6d5cec1b6cda678d2..4b82f913589b699a40c8b2c9a46b3d616fa3d8b7 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -40,6 +40,7 @@ dependencies {
 	        exclude(group = "io.papermc.paper", module = "paper-api") // Paper-API 제거(Paper-API가 Conflict될 위험 존재)
 			exclude(group = "net.java.dev.jna", module = "jna")
             exclude(group = "org.slf4j", module = "slf4j-api")
+			exclude(group = "org.mineacademy.plugin")
     } // Paper-API 제거(Paper-API가 Conflict될 위험 존재)
     implementation("ca.spottedleaf:concurrentutil:0.0.3") // Paper - Add ConcurrentUtil dependency
     // Paper start
